src/FSAL/FSAL_CORTXFS/conf/nfs_setup.sh

```diff
@@ -99,6 +100,35 @@ function motr_lib_init {
 	[ $? -ne 0 ] && die "Failed to Initialise motr_lib fs_meta index"
 }
 
+function prepare_pnfs_conf {
+IFS=',' read -ra ADDR <<< "$pNFS_DATA_SERVER"
+count=0
+rm -rf /etc/ganesha/pNFS.conf
+echo "[pNFS]" >> /etc/ganesha/pNFS.conf
+echo "  pNFS_Role = $pNFS_ROLE" >> /etc/ganesha/pNFS.conf
+
+if [ "$pNFS_ROLE" != "PNFS_DISABLED" ]; then
+pNFS_ENABLED=true;
+echo "pNFS_ENABLED is $pNFS_ENABLED"
+fi
+
+if [ "$pNFS_ROLE" == "MDS" ] || [ "$pNFS_ROLE" == "BOTH" ]; then
+for i in "${ADDR[@]}"; do
+let "count= count+1"
+echo "  DS$count = $i:2049" >> /etc/ganesha/pNFS.conf
+done
+
+if [ "$pNFS_ROLE" == "BOTH" ]; then
```

Comment:
Wouldn't it be double counting , as "BOTH" is already taken care in above stanza ?

Reply:
No, its not.In the first part, we are writing the data servers IP address and incrementing the count every time.Only if role is both, IP address of the current system (hostname -i) is additionally written and the count is incremented.